File: Algorithm/Sorting/Cyclic Sort/cyclic_sort.py
def cyclic_sort(arr):
    for i in range(len(arr)):
        while arr[i] != i + 1:
            arr[arr[i] - 1], arr[i] = arr[i], arr[arr[i] - 1]

arr = [1, 2, 3, 9, 8, 7, 5, 4, 6]
print("Original array:", arr)
cyclic_sort(arr)
print("Array after cyclic sort:", arr)

File: Algorithms/Sorting/Cyclic Sort/cyclicSort.c
#include <stdio.h>
#include "../array.h"

void cyclicSort(int arr[], int n) {
    for (int i = 0; i < n; ) {
        if (arr[i] == i + 1) {
            i++;
        } else {
            swap(&arr[i], &arr[arr[i] - 1]);
        }
        printArray(arr, n);
    }
}

int main() {
    int arr[] = {1, 2, 3, 9, 8, 7, 5, 4, 6};
    int n = sizeof(arr) / sizeof(arr[0]);

    printf("Original array: ");
    printArray(arr, n);

    cyclicSort(arr, n);

    printf("Array after cyclic sort: ");
    printArray(arr, n);

    return 0;
}
